google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryParameterValue.java

```diff
@@ -0,0 +1,439 @@
+  public static QueryParameterValue float64(Double value) {
+    return of(value, StandardSQLTypeName.FLOAT64);
+  }
+
+  /** Creates a {@code QueryParameterValue} object with a type of FLOAT64. */
+  public static QueryParameterValue float64(Float value) {
+    return of(value, StandardSQLTypeName.FLOAT64);
+  }
+
+  /** Creates a {@code QueryParameterValue} object with a type of STRING. */
+  public static QueryParameterValue string(String value) {
+    return of(value, StandardSQLTypeName.STRING);
+  }
+
+  /** Creates a {@code QueryParameterValue} object with a type of BYTES. */
+  public static QueryParameterValue bytes(byte[] value) {
+    return of(value, StandardSQLTypeName.BYTES);
+  }
+
+  /** Creates a {@code QueryParameterValue} object with a type of TIMESTAMP. */
+  public static QueryParameterValue timestamp(Long value) {
+    return of(value, StandardSQLTypeName.TIMESTAMP);
+  }
+
+  /**
+   * Creates a {@code QueryParameterValue} object with a type of TIMESTAMP. Must be in the format
+   * "yyyy-MM-dd HH:mm:ss.SSSSSSZZ", e.g. "2014-08-19 12:41:35.220000+00:00".
+   */
+  public static QueryParameterValue timestamp(String value) {
+    return of(value, StandardSQLTypeName.TIMESTAMP);
+  }
+
+  /**
+   * Creates a {@code QueryParameterValue} object with a type of DATE. Must be in the format
+   * "yyyy-MM-dd", e.g. "2014-08-19".
+   */
+  public static QueryParameterValue date(String value) {
+    return of(value, StandardSQLTypeName.DATE);
+  }
+
+  /**
+   * Creates a {@code QueryParameterValue} object with a type of TIME. Must be in the format
+   * "HH:mm:ss.SSSSSS", e.g. "12:41:35.220000".
+   */
+  public static QueryParameterValue time(String value) {
+    return of(value, StandardSQLTypeName.TIME);
+  }
+
+  /** Creates a {@code QueryParameterValue} object with a type of DATETIME.
+   * Must be in the format "yyyy-MM-dd HH:mm:ss.SSSSSS", e.g. ""2014-08-19 12:41:35.220000". */
+  public static QueryParameterValue dateTime(String value) {
+    return of(value, StandardSQLTypeName.DATETIME);
+  }
+
+  /**
+   * Creates a {@code QueryParameterValue} object with a type of ARRAY, and an array element type
+   * based on the given class.
+   */
+  public static <T> QueryParameterValue array(T[] array, Class<T> clazz) {
+    return array(array, classToType(clazz));
+  }
+
+  /**
+   * Creates a {@code QueryParameterValue} object with a type of ARRAY the given array element type.
+   */
+  public static <T> QueryParameterValue array(T[] array, StandardSQLTypeName type) {
+    List<QueryParameterValue> listValues = new ArrayList<>();
+    for (T obj : array) {
+      listValues.add(QueryParameterValue.of(obj, type));
+    }
+    return QueryParameterValue.newBuilder()
+        .setArrayValues(listValues)
+        .setType(StandardSQLTypeName.ARRAY)
+        .setArrayType(type)
+        .build();
+  }
+
+  private static <T> StandardSQLTypeName classToType(Class<T> type) {
+    if (Boolean.class.isAssignableFrom(type)) {
```

Comment:
Ok, as said the java.sql do have a matching classes but it is fine to not supportit if documented.

I think you would be better of documenting such behavior in the method[s]  `of(T value, Class<T> type)` and `array(T[] array, Class<T> clazz)`  rather than in the class level javadoc.